nautobot/users/tests/test_api.py

```diff
@@ -73,6 +73,87 @@ def setUpTestData(cls):
         Group.objects.create(name="Group 3")
 
 
+class TokenTest(APIViewTestCases.APIViewTestCase):
+    model = Token
+    brief_fields = ["display", "id", "key", "url", "write_enabled"]
+    bulk_update_data = {
+        "description": "New description",
+    }
+
+    def setUp(self):
+        super().setUp()
+
+        tokens = (
+            # We already start with one Token, created by the test class
+            Token(user=self.user),
+            Token(user=self.user),
+        )
+        # Use save() instead of bulk_create() to ensure keys get automatically generated
+        for token in tokens:
+            token.save()
+
+        self.create_data = [
+            {
+                "user": self.user.pk,
+            },
+            {
+                "user": self.user.pk,
+            },
+            {
+                "user": self.user.pk,
+            },
+        ]
+
+    def test_provision_token_valid(self):
+        """
+        Test the provisioning of a new REST API token given a valid username and password.
+        """
+        data = {
+            "username": "user1",
+            "password": "abc123",
+        }
+        user = User.objects.create_user(**data)
+        url = reverse("users-api:token_provision")
+
+        response = self.client.post(url, **self.header, data=data)
```

Comment:
I think for these tests we would want to omit `**self.header` as that is mostly used to include the pre-defined user's token as an `Authorization: Token ...` header, which should *not* be a part of these requests.